wsd-frontend/src/components/wsd/Meme/client/MemeThreeDotMenu.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import * as Icons from 'lucide-react'
+
+import { Button } from '@/components/shadcn/button'
+import { Overlay, OverlayContent, OverlayTrigger } from '@/components/shadcn/overlay'
+
+import { APIType, Includes } from '@/api'
+import { downloadAndFormatImage } from '@/lib/fileUtils'
+
+import { toast } from 'sonner'
+
+export default function MemeThreeDotMenu({
+  post,
+}: {
+  post: Includes<Includes<APIType<'Post'>, 'user', APIType<'User'>>, 'tags', APIType<'PostTag'>[]>
+}) {
+  const actions = [
+    {
+      label: 'Download',
+      icon: Icons.Download,
+      action: async () => {
+        try {
+          await downloadAndFormatImage(post.image, `${post.title}.png`, 'image/png', 1)
+          toast('Image downloaded successfully')
+        } catch {
+          toast('Unknown error downloading image')
+        }
+      },
+    },
+  ]
+
+  return (
+    <Overlay breakpoint="md">
+      <OverlayTrigger>
+        <a
```

Comment:
Why is this an anchor but not a button anymore? It loses some semantic value and some default css coming from shadcn if we change this to an anchor, unless you have a specific reason can you change this back to a button?